pkg/streamingpromql/testdata/ours/aggregators.test

```diff
@@ -31,3 +31,55 @@ load 1m
 eval range from 1m to 1m30s step 1s sum(some_metric_with_staleness)
   # Should return no results.
 
+clear
+
+# Test native histogram aggregations
+load 1m
+	single_histogram{label="value"} {{schema:0 sum:2 count:4 buckets:[1 2 1]}} {{sum:2 count:4 buckets:[1 2 1]}}
+	single_histogram{label="value2"} {{schema:1 sum:5 count:5 buckets:[1 3 1]}} {{schema:0 sum:2 count:4 buckets:[1 2 1]}}
+	single_histogram{label="value2", another="value"} {{schema:1 sum:10 count:9 buckets:[3 3 3]}} {{schema:2 sum:2 count:4 buckets:[1 2 1]}}
+
+eval instant at 0m sum(single_histogram)
+	{} {{schema:0 sum:17 count:18 buckets:[5 12 1]}}
+
+eval instant at 0m sum by (label) (single_histogram)
+	{label="value"} {{schema:0 count:4 sum:2 buckets:[1 2 1]}}
+	{label="value2"} {{schema:1 count:14 sum:15 buckets:[4 6 4]}}
+
+eval range from 0 to 1m step 1m  sum(single_histogram)
+	{} {{schema:0 sum:17 count:18 buckets:[5 12 1]}} {{schema:0 sum:6 count:12 buckets:[3 7 2]}}
+
+clear
+
+# Test a mix of float and histogram values
+load 1m
+	single_histogram{label="value"}                    0 1 {{schema:0 sum:2 count:4 buckets:[1 2 1]}}  {{sum:2 count:4 buckets:[1 2 1]}}          2
+	single_histogram{label="value2"}                   0 2 {{schema:1 sum:5 count:5 buckets:[1 3 1]}}  {{schema:0 sum:2 count:4 buckets:[1 2 1]}} 4
+	single_histogram{label="value2", another="value"}  0 3 {{schema:1 sum:10 count:9 buckets:[3 3 3]}} {{schema:2 sum:2 count:4 buckets:[1 2 1]}} 6
+
+# TODO: Verify if these tests are correct. At the moment they match promql engine.
+# See: https://github.com/prometheus/prometheus/issues/14172
+
+# What I would expect
+# eval range from 0 to 4m step 1m  sum by (label) (single_histogram)
+#	{label="value"}  0 1 {{count:4 sum:2 buckets:[1 2 1]}}            {{sum:2 count:4 buckets:[1 2 1]}}          2
+#	{label="value2"} 0 5 {{schema:1 count:14 sum:15 buckets:[4 6 4]}} {{schema:2 count:8 sum:4 buckets:[4 6 4]}} 10
+
+# What both engines return
+eval range from 0 to 4m step 1m  sum by (label) (single_histogram)
+	{label="value"}  0 1 1 1 2
+	{label="value2"} 0 5 5 5 10
+
+clear
+
+# Test a mix of float and histogram values at the same point
+load 1m
+	single_histogram{label="value"}   0 1                                          {{schema:0 sum:2 count:4 buckets:[1 2 1]}} {{sum:2 count:4 buckets:[1 2 1]}} 2 {{sum:2 count:4 buckets:[1 2 1]}}
+	single_histogram{label="value2"}  0 {{schema:1 sum:5 count:5 buckets:[1 3 1]}} {{schema:0 sum:2 count:4 buckets:[1 2 1]}} 4                                 6 {{sum:2 count:4 buckets:[1 2 1]}}
+
+# If a float is present, the histogram is ignored.
+# If a float comes after a histogram, a lookback'd float is used instead of the histogram (see: https://github.com/prometheus/prometheus/issues/14172)
+eval range from 0 to 5m step 1m  sum(single_histogram)
+	{} 0 1 1 5 8 {{sum:4 count:8 buckets:[2 4 2]}}
```

Comment:
What happens if we have two series, one with a single float sample and another with a single histogram sample? That should help us exercise this scenario without running into the other issue you've discovered.

Based on https://prometheus.io/docs/prometheus/latest/querying/operators/#operators-for-native-histograms, I'd expect no points to be returned at all in this case: "If either operator has to aggregate a mix of histogram samples and float samples, the corresponding vector element is removed from the output vector entirely."

Reply:
I expect it to return just the sum of floats (despite the docs). This is what is exercised here, but let me confirm in code.

Reply:
Looks like I was wrong. The promql engine returns nothing as documented. Our engine currently returns both a float and histogram, and is therefore wrong. Will need to fix that.